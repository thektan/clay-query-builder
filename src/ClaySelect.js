import React from 'react';
import {PropTypes} from 'prop-types';

class ClaySelect extends React.Component {
    render() {
        const {
            selected, 
            options,
            className,
            ...otherProps
        } = this.props;

        return (
            <select className={`form-control ${className}`} {...otherProps} value={selected}>
                {options.map(
                    (option, index) => (
                        <option value={option.value} key={index}>{option.label}</option>
                    )
                )}
            </select>
        );
    }
}

ClaySelect.propTypes = {
    selected: PropTypes.string,
    options: PropTypes.array.isRequired,
    onChange: PropTypes.func.isRequired
}

export default ClaySelect;