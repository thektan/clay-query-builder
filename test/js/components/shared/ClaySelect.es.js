import React from 'react';
import ClaySelect from 'components/shared/ClaySelect.es';
import {cleanup, render} from 'react-testing-library';

describe(
	'ClaySelect',
	() => {
		afterEach(cleanup);

		it(
			'should render',
			() => {
				const {asFragment} = render(
					<ClaySelect onChange={jest.fn()} options={[]} />
				);

				expect(asFragment()).toMatchSnapshot();
			}
		);

		it(
			'should render with an item selected',
			() => {
				const {container} = render(
					<ClaySelect
						onChange={jest.fn()}
						options={[
							{
								label: 'aLabel',
								value: 'aValue'
							},
							{
								label: 'aLabel',
								value: 'bValue'
							}
						]}
						selected="bValue"
					/>
				);

				expect(container.firstChild.value).toEqual('bValue');
			}
		);
	}
);