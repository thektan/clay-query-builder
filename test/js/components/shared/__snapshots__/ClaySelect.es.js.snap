// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaySelect should render 1`] = `
<DocumentFragment>
  <select
    class="form-control"
  />
</DocumentFragment>
`;
